common/components/BalanceSidebar/TokenBalances/AddCustomTokenForm/AddressField.tsx

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import { Input } from 'components/ui';
+import { translateRaw } from 'translations';
+import { IGenerateAddressLookup } from './AddCustomTokenForm';
+import { isValidETHAddress } from 'libs/validators';
+import { Result } from 'nano-result';
+
+interface OwnProps {
+  addressLookup: IGenerateAddressLookup;
+  onChange(address: Result<string>): void;
+}
+
+enum ErrType {
+  INVALIDADDR = 'Not a valid address',
+  ADDRTAKEN = 'A token with this address already exists'
+}
+
+interface State {
+  address: Result<string>;
+  userInput: string;
+}
+
+export class AddressField extends React.Component<OwnProps, State> {
+  public state: State = {
+    address: Result.from({ res: '' }),
```

Comment:
Just a thought, might be helpful to have a `Result.init` method or something to act as a placeholder in situations like this.

Reply:
Yeah, it wouldn't save much on the boilerplate though since i'd then have to specify the type of the generic as `<string>` anyway.